cosmos/config.py

```diff
@@ -40,6 +40,13 @@ class RenderConfig:
     exclude: list[str] = field(default_factory=list)
     dbt_deps: bool = True
     node_converters: dict[DbtResourceType, Callable[..., Any]] | None = None
+    dbt_executable_path: str | Path = get_system_dbt()
+    dbt_project_path: InitVar[str | Path | None] = None
+
+    project_path: Path | None = field(init=False)
```

Comment:
I was experimenting with a feature of data classes here.
When the user creates an object here, they use dbt_project_path, but dbt_project_path is not an attribute of the class once constructed

dbt_project_path is only used for object generation and is passed directly to the __post_init__

project_path on the other hand, cant be defined using the init, but, it can be accessed as an object property, which, to your point will guarantee it being Path or None, simiar to what i did in ProjectConfig, without requiring a custom constructor.